.github/workflows/precommit.yml

```diff
@@ -17,6 +17,26 @@ jobs:
           java-version: ${{ matrix.java }}
           distribution: temurin
           cache: gradle
+      - name: Set up JDK 17
```

Comment:
Take it or leave it: might be useful to migrate this setup into an inner action that looks very similar to `actions/setup-java` which handles all of the edge cases.

Example from the security repo: https://github.com/opensearch-project/security/tree/main/.github/actions